packages/styled-components/src/sheet/Rehydration.js

```diff
@@ -53,15 +52,12 @@ const rehydrateNamesFromContent = (sheet: Sheet, id: string, content: string) =>
 };
 
 const rehydrateSheetFromTag = (sheet: Sheet, style: HTMLStyleElement) => {
-  const rawHTML = style.innerHTML;
+  const parts = style.innerHTML.split(SPLITTER);
   const rules: string[] = [];
-  let parts;
-
-  // parts = [match, selector, content]
-  // eslint-disable-next-line no-cond-assign
-  while ((parts = RULE_RE.exec(rawHTML))) {
-    const marker = parts[1].match(MARKER_RE);
 
+  for (let i = 0, l = parts.length; i < l; i++) {
+    const part = parts[i].trim();
+    const marker = part.match(MARKER_RE);
     if (marker) {
```

Comment:
Let's add a `continue` shortcircuit at the beginning if `part` is empty. Since we're using the `split` method it's possible there might be crud in there